### ISSUE TITLE
[Bug] Auto-document for `externalDocs` Bug

### ISSUE THREAD

**Issue by ChihweiLHBird** [sanic-org/sanic-ext#37]
```python
app = Sanic("test")

@app.route("/test1")
@openapi.document(
    ExternalDocumentation("http://example.com/mor1111", "Find more info her111e")
)
async def handler1(request: Request):
    return text("ok")

@app.route("/test2")
async def handler2(request: Request):
    """
    openapi:
    ---
    summary: This is a summary.
    externalDocs:
        description: Find more info here
        url: http://example.com/more
    """
    return text("ok")

app.run()
```

**The `externalDocs` field of path `/test1` is incorrect:**

```json
{
    "openapi": "3.0.0",
    "info": {
        "title": "API",
        "version": "1.0.0",
        "contact": {}
    },
    "paths": {
        "\/test1": {
            "get": {
                "operationId": "get~handler1",
                "summary": "Handler1",
                "externalDocs": {
                    "url": {
                        "url": "http:\/\/example.com\/mor1111",
                        "description": "Find more info her111e"
                    },
                    "description": null
                },
                "responses": {
                    "default": {
                        "description": "OK"
                    }
                }
            }
        },
        "\/test2": {
            "get": {
                "operationId": "get~handler2",
                "summary": "This is a summary.",
                "externalDocs": {
                    "description": "Find more info here",
                    "url": "http:\/\/example.com\/more"
                },
                "responses": {
                    "default": {
                        "description": "OK"
                    }
                }
            }
        }
    },
    "tags": [],
    "servers": [],
    "security": []
}
```

**Comment by ahopkins** [comment]
What am I looking at? They look the same to me. What am I missing?

**Comment by ChihweiLHBird** [comment]
> What am I looking at? They look the same to me. What am I missing?

```json
"externalDocs": {
    "url": {
        "url": "http:\/\/example.com\/mor1111",
        "description": "Find more info her111e"
    },
    "description": null
},
```

VS

```json
"externalDocs": {
    "description": "Find more info here",
    "url": "http:\/\/example.com\/more"
},
````

😂

**Comment by ahopkins** [comment]
Got it. Thanks.

**Comment by ChihweiLHBird** [comment]
> Got it. Thanks. 

Sorry, my bad. It was test1

**Comment by ahopkins** [comment]
FYI, if you write `Closes #37`, `Fixes #37`, `Resolves #37` or something like that in the description for the PR, then GitHub will automatically close this issue.
